### PR TITLE
First Version

### PR DESCRIPTION
**Pull request by ogomezso**
- Json and AVRO (value schema as String) supported
- Use Request xml file to create SOAP request and execute call

TODO:

-  Separate Task and Connector Config
- Allow more than a Request (on separate tasks)
- Limit task to number of request file
- Support for SOAP Envelope Creation from remote WSDL
- Allow full qualified AVRO Schemas

### CODE REVIEW

File: src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import jakarta.xml.soap.SOAPPart;
+import jakarta.xml.ws.Dispatch;
+import jakarta.xml.ws.Service;
+import jakarta.xml.ws.soap.SOAPBinding;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class SoapClient {
+
+  private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(4);
```
Comment (nachomdo) [inline src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java:46]: consider retrieving the pool size dynamically from the runtime or configuration. Also double check that you are following executor sizing best practices  for CPU vs IO bound workloads https://gist.github.com/djspiewak/46b543800958cf61af6efa8e072bfd5c

File: src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    MessageFactory messageFactory = MessageFactory.newInstance();
+    SOAPMessage message = messageFactory.createMessage();
+    SOAPPart soapPart = message.getSOAPPart();
+    StreamSource preppedMsgSrc = new StreamSource(new FileInputStream(
+        pathToMessage));
+    soapPart.setContent(preppedMsgSrc);
+    message.saveChanges();
+
+    return dispatch.invoke(message);
```
Comment (nachomdo) [inline src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java:98]: consider some connection and socket level timeouts. Are you relying on the poll interval for retrying?  maybe adding some exponential backoff and jitter to retries.

File: src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        .get(pollInterval + 5L,
+            TimeUnit.SECONDS);
+    log.debug("result: " + result.toString());
+    log.debug("get future at : " + LocalDateTime.now());
+    return result;
+  }
+
+  private SOAPMessage invoke(QName serviceName, QName portName, String endpointUrl,
+      String soapActionUri, File pathToMessage) throws SOAPException, FileNotFoundException {
+    Service service = Service.create(serviceName);
```
Comment (nachomdo) [inline src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java:82]: are we creating a new connection every time we poll? If so maybe consider some resource pooling over keep alive connections.

File: src/main/java/com/github/ogomezso/kafka/connect/soap/source/SoapSourceConnectorConfig.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.Type;
+
+import com.github.jcustenborder.kafka.connect.utils.config.ConfigUtils;
+
+public class SoapSourceConnectorConfig extends AbstractConfig {
+
+  public static final String ENDPOINT_URL = "endpointUrl";
```
Comment (nachomdo) [inline src/main/java/com/github/ogomezso/kafka/connect/soap/source/SoapSourceConnectorConfig.java:31]: consider to expose a CONNECTION_TIMEOUT configuration.

**Review by nachomdo — COMMENTED** [review]
Looks good  in general 👍  I put some suggestions but they are in most cases just optimisations 😅

File: src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java (hunk 1 above)
Comment (ogomezso) [reply src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java:46]: DONE. 

Retrieving poolsize on runtime and assign half to the thread pool size

File: src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java (hunk 3 above)
Comment (ogomezso) [reply src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java:125]: On TODO list for further PR (I'm gonna use Apache Commons Object pool as we discussed)

File: src/main/java/com/github/ogomezso/kafka/connect/soap/source/SoapSourceConnectorConfig.java (hunk 4 above)
Comment (ogomezso) [reply src/main/java/com/github/ogomezso/kafka/connect/soap/source/SoapSourceConnectorConfig.java:30]: now in connector config with default value

File: src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java (hunk 2 above)
Comment (ogomezso) [reply src/main/java/com/github/ogomezso/kafka/connect/soap/client/SoapClient.java:144]: Circuit Breaker added at task level

**Pull request merged**